### DISCUSSION TITLE
CreatableSelect with onCreateOption prop has a bug

### DISCUSSION THREAD

**Discussion by t6adev** [csandman/chakra-react-select#112] [Q&A] [answered]
**Describe the bug**
If you use CreatableSelect with onCreateOption, `<CreatableSelect onCreateOption={...} />` to create a new option, the created value is not displayed.

**Link to Reproduction**
https://codesandbox.io/s/chakra-react-select-oncreateoption-cy2ko

You can confirm that react-select has no problem with that.
https://codesandbox.io/s/z19sm
https://react-select.com/creatable

**To Reproduce**

1. Go to the reproduction link
2. Type anything to create an option then enter
3. The inputted value is not displayed

**Expected behavior**
The created value should be displayed.

Thanks.

**Comment by csandman** [comment, marked as answer]
So if you look at the documentation for that prop, this is what it says:

> `onCreateOption`
> 
> ---
> 
> If provided, this will be called with the input value when a new option is created, and `onChange` will **not** be called. Use this when you need more control over what happens when new options are created.
> 
> (`inputValue:string`) => `void`
>
> — https://react-select.com/props#creatable-props

So this means that if you use this option, you will have to implement some manual code for setting the option. If you look at [the code in the example you provided for the original package](https://codesandbox.io/s/z19sm?file=/example.tsx), you'll see that in their `onCreateOption` function they're handling how the new value is being set on the select.

I made a modified version of the `chakra-react-select` demo you made implementing all of the code from the original example, and it works in the same way: https://codesandbox.io/s/chakra-react-select-oncreateoption-forked-q7703?file=/app.tsx

Let me know if this solves your issue!

**Comment by t6adev** [comment]
Oh, So sorry for bothering you, it's my fault.
I didn't confirm the doc and my shared code carefully.

I also re-implemented my sample code then it works well.
https://codesandbox.io/s/chakra-react-select-oncreateoption-cy2ko?file=/app.tsx

Thank you so much for taking your time.

**Comment by csandman** [comment]
No worries, glad I could help!
